src/test/compile-fail/issue-22034.rs

```diff
@@ -14,6 +14,6 @@ fn main() {
     let foo: *mut libc::c_void;
     let cb: &mut Fn() = unsafe {
         &mut *(foo as *mut Fn())
-        //~^ ERROR use of possibly uninitialized variable: `foo`
+        //~^ ERROR illegal cast
```

Comment:
Looks like this test case was actually testing the wrong thing. The associated issue is actually just another instance of casting thin <-> fat pointers, and this test case should not have closed that issue. Note that the error is complaining about something completely unrelated to why the code is actually invalid.

It would be great if you could actually fix up this test case to avoid the other reasons why this case is invalid. A corrected test case looks like

```rust
fn main() {
    let ptr: *mut () = 0 as *mut _;
    let _: &mut Fn() = unsafe {
        &mut *(ptr as *mut Fn())
    };
}
```